### PR TITLE
Update DG 

### PR DESCRIPTION
**Pull request by GlendaChong**
- Update implementation details for Add and Filter features

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -445,26 +436,17 @@ The following activity diagram will show how `RemarkCommand` can achieve the fun
...
-
-4. The `AddCommandParser` creates a `Company` object, and then creates an `AddCommand` object with the `Company` object.
-
-5. The `AddCommand` object can communicate with the `Model` when it is executed. It first checks if there's a duplicate input, which has the same company name, application role and application deadline.
-
-6. If the `Model` does not have a duplicate, the `AddCommand` object calls `Model#addCompany` to add the new `Company` into LinkMeIn.
+The `add` feature is implemented using the `AddCommand` class. The `AddCommand` object takes in a `Company` object. Only if all the inputs for the parameters are valid and all compulsory parameters are present, then the `Company` object is created.
 
-7. Finally, the `AddCommand` object returns the `CommandResult` object.
+The `add` feature includes the following operations in `ModelManager`, which is implemented by the `Model` interface:
```
Comment (alientian) [inline docs/DeveloperGuide.md:444]:
```suggestion
The `add` feature includes the following operations in `ModelManager`, which implements the `Model` interface:
```
Model is the superclass of ModelManager

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -290,22 +290,13 @@ With the design considerations, we've chosen the alternatives that provide a bal
 
 ### Filter Feature
 
-The `filter` command allows users to filter the company list by the application status.
+The `filter` command allows users to filter the list of companies based on the valid application status. 
 
 #### Implementation
 
-The `filter` command uses the `ApplicationStatusPredicate` class, which tests and returns true if a company's application status matches the application status input specified by the user. The `ApplicationStatusPredicate` class implements the `Predicate` interface, which allows it to be used in the `Model` interface's `updateFilteredCompanyList(Predicate<Company> predicate)` method.
+The `filter` command uses a new predicate, `ApplicationStatusPredicate`, which tests and returns true if a company's application status matches the application status input specified by the user. The `ApplicationStatusPredicate` class implements the `Predicate` interface, which allows it to be used in the `Model` interface's `updateFilteredCompanyList(Predicate<Company> predicate)` method to filter the list of companies. All companies that match the application status input will be displayed in the updated `CompanyListPanel`.
```
Comment (alientian) [inline docs/DeveloperGuide.md:297]:
```suggestion
The `filter` command uses a new predicate, `ApplicationStatusPredicate`, which tests and returns true if a company's application status matches the application status input specified by the user. The `ApplicationStatusPredicate` class implements the `Predicate` interface, which allows it to be used in the `Model#updateFilteredCompanyList(Predicate<Company> predicate)` method to filter the list of companies. All companies that match the application status input will be displayed in the updated `CompanyListPanel`.
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -473,7 +455,7 @@ The following sequence diagram illustrates how the `add` command works:
 The lifeline for `AddCommandParser` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
 </div>
 
-The following activity diagram shows how the `add` command works:
+The following activity diagram shows what happens when a `user` execute the `add` command.
```
Comment (tayruxin) [inline docs/DeveloperGuide.md:458]: wait is there a need for user to be `user` instead of user

Comment (GlendaChong) [reply docs/DeveloperGuide.md:458]: oh yes, my mistake will fix it thanks

**Pull request merged**